File: SimpleLinearRegression/EJ_simple_linear_regression.py
#Simple Linear Regression

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#import the dataset
dataset = pd.read_csv('MachineLearningA-Z/SimpleLinearRegression/Salary_Data.csv')

#create matrix for independent and dependent variables separately
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

#Partitioning Data
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=.33,random_state=0)

#Train the Simple Linear Regression Model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)
regressor.coef_

#Predicting the Test set results
#will create a vector of the test set salaries
y_pred = regressor.predict(X_test)

#Visualize training set results
plt.scatter(X_train,y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color="blue")
plt.title('Salary vs. Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#Visualize training set results
plt.scatter(X_test,y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color="blue")
plt.title('Salary vs. Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: SimpleLinearRegression/EJ_simple_linear_regression.R
#Simple Linear Regression

library(ggplot2)

#Read in data
dataset <- read.csv("SimpleLinearRegression/Salary_Data.csv")
# dataset <- dataset[,2:3]

#Partition the data
library(caTools)
set.seed(123)

split = sample.split(dataset$Salary, SplitRatio = 2/3)

training_set = subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)

#Train the model
regressor <- lm(Salary~YearsExperience,data=dataset)

#Predicting Test Set Results
y_pred <- predict(regressor,newdata=test_set)

#Visualize Training set results
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),col='red') + 
  geom_line(aes(x = training_set$YearsExperience, y = 
                  predict(regressor, newdata = training_set)),col='blue') + 
  ggtitle('Salary v Experience (Training Set)') + 
  xlab('Years of Experience') + ylab('Salary')

#Visualize Test set results
ggplot() + geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary)
                      ,col='red') + 
  geom_line(aes(x = training_set$YearsExperience, 
                y = predict(regressor, newdata = training_set)) ,col='blue') + 
  ggtitle('Salary v Experience (Test Set)') + xlab('Years of Experience') + ylab('Salary')

